seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py

```diff
@@ -0,0 +1,159 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+        Linda Xiang <linda.xiang@oicr.on.ca>
+        Edmund Su <linda.xiang@oicr.on.ca>
+"""
+
+
+import os
+from xmlrpc.client import boolean
+from base_checker import BaseChecker
+import subprocess
+import re
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata, skip=False):
+        super().__init__(
+            ctx,
+            metadata=metadata,
+            checker_name=__name__,
+            depends_on=[
+                "c608_bam_sanity",
+                "c610_rg_id_in_bam",
+                "c608_bam_sanity",
+                "c620_submitter_read_group_id_match",
+                "c630_rg_id_in_bam_match",
+                "c670_rg_is_paired_in_bam"
+            ],
+            skip=skip)
+
+    @BaseChecker._catch_exception
+    def check(self):
+        # status already set at initiation
+        if self.status:
+            return
+
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(f'[{self.checker}] {message}')
+            self.message = message
+            self.status = 'INVALID'
+            return
+    
+            
+        offending_ids = []
+        query_bams={}
+
+        for rg in self.metadata.get("read_groups"):
+            if rg['file_r1'].endswith('.bam') and rg['file_r1'] not in query_bams:
+                if 'is_paired_end' not in rg:
+                    message = "Field 'is_pair_end' is not found for readgroup : %s" % rg['submitter_read_group_id']
+                    self.logger.info(f'[{self.checker}] {message}')
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                elif rg['is_paired_end']==None:
+                    message = "Field 'is_pair_end' for readgroup is null. Must be boolean: %s" % rg['submitter_read_group_id']
+                    self.logger.info(f'[{self.checker}] {message}')
+                    self.message = message
+                    self.status = 'INVALID'
+                    return              
+                else:
+                    query_bams[rg['file_r1']]={}
+                    query_bams[rg['file_r1']]['is_paired_end']=rg['is_paired_end']
+                    query_bams[rg['file_r1']]['rg']=[]
+                    query_bams[rg['file_r1']]['rg'].append(rg['read_group_id_in_bam'])
+            elif rg['file_r1'] in query_bams:
+                query_bams[rg['file_r1']]['rg'].append(rg['read_group_id_in_bam'])
+                             
```

Comment:
Do we have a check in place to do existence checks for fields like `read_groups` and `is_paired_end`? We can for sure do this kind of house keeping checks in separate checks.

Reply:
Removed redundant tests now that dependancies are inplace
https://github.com/icgc-argo/seq-tools/blob/19678023562709ee5c26142959b7d4850068b08e/seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py#L61-69